compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -170,7 +170,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 }
             }
 
-            Drop { place, target, unwind, replace: _ } => {
+            Drop { place, target, unwind, replace: _, drop: _, async_fut: _ } => {
```

Comment:
There's something strange going on that's not explained in the comments in `syntax.rs` -- why are these things *entirely ignored* by the interpreter?

Reply:
Yes, both `drop` and `async_fut` fields are only used in compiler/rustc_mir_transform/src/coroutine.rs, StateTransform pass. In `expand_async_drops` async drops are expanded into one or two yield points with poll ready/pending switch.

Drop terminator comments updated.

Reply:
In that case codegen and the interpreter should `bug!` when those fields are present.

Though it seems to me it would probably be better to make this a different terminator. They have very different operational behavior, after all.